src/ex6/server/client/src/components/AddTask/AddTask.jsx

```diff
@@ -0,0 +1,121 @@
+import React, { useEffect, useState } from "react"
+import "./style.css"
+import { ItemClient } from "./item_client"
+import deleteIcon from "./images/delete_icon.svg"
+
+export const AddTask = () => {
+  const itemClient = new ItemClient()
+  const [task, setTask] = useState("")
+  const [tasks, setTasks] = useState([])
+  const [isLoading, setIsLoading] = useState(false)
+  const [error, setError] = useState(null)
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setIsLoading(true)
+      try {
+        const result = await itemClient.getItems()
+        setTasks(result)
+      } catch (error) {
+        setError(error)
+      }
+      setIsLoading(false)
+    }
+    fetchData()
+  }, [])
+
+  console.log("tasks", tasks)
+  const handleSubmit = async (e) => {
+    e.preventDefault()
+    setIsLoading(true)
+    try {
+      const result = await itemClient.postItem(task)
+      setTasks([...tasks, result])
+      setTask("")
+    } catch (error) {
+      setError(error)
+    }
+    setIsLoading(false)
+  }
+
+  const deleteItem = async (id) => {
+    setIsLoading(true)
+    try {
+      const result = await itemClient.deleteItem(id)
+      setTasks(tasks.filter((task) => task.id !== id))
+    } catch (error) {
+      setError(error)
+    }
+    setIsLoading(false)
+  }
+
+  const doneItem = async (id) => {
+    setIsLoading(true)
+    try {
+      const result = await itemClient.toggleDone(id)
+      setTasks(tasks.map((task) => (task.id === id ? result : task)))
+    } catch (error) {
+      setError(error)
+    }
+    setIsLoading(false)
+  }
+
+  const _createItemTextElement = (item) => {
+    return (
+      <div className="item-text">
```

Comment:
also - avoid wrapping elements in divs if it serves no purpose
 in this case you can use the <p> element directly